internal/sys/fs.go

```diff
@@ -56,209 +56,6 @@ type FileEntry struct {
 	File fsapi.File
 }
 
-const direntBufSize = 16
-
-// Readdir is the status of a prior fs.ReadDirFile call.
-type Readdir struct {
-	// cursor is the current position in the buffer.
-	cursor uint64
-
-	// countRead is the total count of files read including Dirents.
-	//
-	// Notes:
-	//
-	// * countRead is the index of the next file in the list. This is
-	//   also the value that Cookie returns, so it should always be
-	//   higher or equal than the cookie given in Rewind.
-	//
-	// * this can overflow to negative, which means our implementation
-	//   doesn't support writing greater than max int64 entries.
-	//   countRead uint64
-	countRead uint64
-
-	// dirents is a fixed buffer of size direntBufSize. Notably,
-	// directory listing are not rewindable, so we keep entries around in case
-	// the caller mis-estimated their buffer and needs a few still cached.
-	//
-	// Note: This is wasi-specific and needs to be refactored.
-	// In wasi preview1, dot and dot-dot entries are required to exist, but the
-	// reverse is true for preview2. More importantly, preview2 holds separate
-	// stateful dir-entry-streams per file.
```

Comment:
the note no longer appears in `windowedReaddir` because it no longer applies: the same effect is obtained by decorating the `windowedReaddir` with a `concatReaddir`; this `concatReaddir` tacks a `sliceReaddir` with the dot entries before a `windowedReaddir` that fetches the directory listing (see `LookupReaddir`).

🤔 maybe better turn this into a comment to that method.